proxy/src/main/java/org/apache/rocketmq/proxy/service/route/MessageQueueSelector.java

```diff
@@ -154,6 +171,82 @@ public AddressableMessageQueue selectOne(boolean onlyBroker) {
         return selectOneByIndex(nextIndex, onlyBroker);
     }
 
+    public AddressableMessageQueue selectOneByPipeline(boolean onlyBroker) {
+        if (topicRouteService.getMqFaultStrategy().isSendLatencyFaultEnable() && topicRouteService != null) {
```

Comment:
TopicRouteService could be null if it is not passed into the constructor.
Moreover, if the constructor without topicRouteService is no longer used, we can delete it.

Reply:
Yes, I have modified as you said. You can check commits history.
Thank you!